grove/connectors/bigquery/query.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+"""Google BigQuery connector for Grove."""
+
+import json
+from datetime import datetime, timedelta, timezone
+
+from google.auth.exceptions import GoogleAuthError
+from google.oauth2 import service_account
+from google.cloud import bigquery
+
+from grove.connectors import BaseConnector
+from grove.constants import CHRONOLOGICAL
+from grove.exceptions import (
+    ConfigurationException,
+    NotFoundException,
+    RequestFailedException,
+)
+
+def as_bigquery_timestamp(epoch_ms_str: str) -> str:
+    """
+    Converts a string containing epoch time in milliseconds to a BigQuery-compatible timestamp string.
+
+    :param epoch_ms_str: The epoch time in milliseconds as a string.
+    :return: A BigQuery TIMESTAMP formatted date string (YYYY-MM-DD HH:MM:SS+00).
+    """
+    dt = datetime.fromtimestamp(int(epoch_ms_str) / 1000.0, tz=timezone.utc)
+    # BigQuery expects "+00" not "+0000" or "+00:00"
+    return dt.strftime("%Y-%m-%d %H:%M:%S+00")
+
+
+class Connector(BaseConnector):
+    CONNECTOR = "bigquery_query"
+    POINTER_PATH = "gmail.event_info.timestamp_usec"
+    LOG_ORDER = CHRONOLOGICAL
+
+    def collect(self):
+        """Collects logs from the specified table using Google BigQuery API."""
+        self.logger.info("Starting data collection from BigQuery.")
+
+        try:
+            project_id = self.configuration.project_id
+            dataset_name = self.configuration.dataset_name
+            table_name = self.configuration.table_name
+            columns = self.configuration.columns
+            POINTER_PATH = self.configuration.pointer_path
+
+
+            if not POINTER_PATH:
+                raise ConfigurationException(
+                    "POINTER_PATH is not set in the configuration."
+                )
+
+            if not all(isinstance(value, str) for value in [project_id, dataset_name, table_name]):
+                raise ConfigurationException(
+                    "project_id, dataset_name, and table_name must all be strings."
+                )
```

Comment:
Minor: Removal of the in-line logic.

```suggestion
            for value in [project_id, dataset_name, table_name]:
                if not isinstance(value, str):
                    raise ConfigurationException(f"{value} must be a string")
```

Reply:
fixed in [b183f4c](https://github.com/hashicorp-forge/grove/pull/83/commits/b183f4cbbd5178b18423ef614dc62263684d2a6b)